### PR TITLE
Do not add strings to xml schema

### PR DESCRIPTION
**Pull request by LingxiaChen**
Don't call the XmlSchemaCollection.Add method whose second parameter's type is string.

### CODE REVIEW

File: src/Microsoft.NetCore.Analyzers/Core/Security/DoNotAddSchemaByURL.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                {
+                    return;
+                }
+
+                compilationStartAnalysisContext.RegisterOperationAction(operationAnalysisContext =>
+                {
+                    var invocationOperation = (IInvocationOperation)operationAnalysisContext.Operation;
+                    var methodSymbol = invocationOperation.TargetMethod;
+
+                    if (methodSymbol.Name == "Add" &&
```
Comment (mavasani) [inline src/Microsoft.NetCore.Analyzers/Core/Security/DoNotAddSchemaByURL.cs]: Did you miss a check `methodSymbol.ContainingType.Equals(xmlSchemaCollectionTypeSymbol)`?

Comment (mavasani) [reply src/Microsoft.NetCore.Analyzers/Core/Security/DoNotAddSchemaByURL.cs]: Also suggest add a unit test for the same.

File: src/Microsoft.NetCore.Analyzers/UnitTests/Security/DoNotAddSchemaByURLTests.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+using Xunit;
+
+namespace Microsoft.NetCore.Analyzers.Security.UnitTests
+{
+    public class DoNotAddSchemaByURLTests : DiagnosticAnalyzerTestBase
+    {
+        [Fact]
+        public void TestAddWithStringStringParametersDiagnostic()
+        {
+            VerifyCSharp(@"
```
Comment (mavasani) [inline src/Microsoft.NetCore.Analyzers/UnitTests/Security/DoNotAddSchemaByURLTests.cs]: Can you add VisualBasic tests as well?

File: src/Microsoft.NetCore.Analyzers/Core/Security/DoNotAddSchemaByURL.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Security
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class DoNotAddSchemaByURL : DiagnosticAnalyzer
+    {
+        internal const string DiagnosticId = "CA5366";
```
Comment (dotpaul) [inline src/Microsoft.NetCore.Analyzers/Core/Security/DoNotAddSchemaByURL.cs]: Can you call this one "CA3061"?  There used to be an FxCop rule that was trying to do the same thing.

File: src/Microsoft.NetCore.Analyzers/Core/Security/SystemSecurityCryptographyResources.resx [hunk 4]
```diff
@@ -126,6 +126,15 @@
   <data name="ApprovedCipherModeMessage" xml:space="preserve">
     <value>It uses an unsafe Cipher Mode {0}</value>
   </data>
+  <data name="DoNotAddSchemaByURL" xml:space="preserve">
+    <value>Do Not Add Schema By URL</value>
+  </data>
+  <data name="DoNotAddSchemaByURLDescription" xml:space="preserve">
+    <value>This overload of XmlSchemaCollection.Add method is potentially unsafe because it may load untrusted XML which includes dangerous external references. Either use an XmlSecureResolver or disable DTD processing.</value>
```
Comment (dotpaul) [inline src/Microsoft.NetCore.Analyzers/Core/Security/SystemSecurityCryptographyResources.resx]: Is there a way for people to disable DTD processing or specify an XML resolver with XmlSchemaCollection.Add(string, string)?  I vote we just get rid of the description entirely.

File: src/Microsoft.NetCore.Analyzers/Core/Security/SystemSecurityCryptographyResources.resx [hunk 5]
```diff
@@ -126,6 +126,15 @@
...
+  </data>
+  <data name="DoNotAddSchemaByURLMessage" xml:space="preserve">
+    <value>This overload of the Add method is potentially unsafe because it may resolve dangerous external references</value>
```
Comment (dotpaul) [inline src/Microsoft.NetCore.Analyzers/Core/Security/SystemSecurityCryptographyResources.resx]: I think we should use the old FxCop message, or maybe just the first two sentences:

Do not use unsafe overloads of System.Xml.Schema.XmlSchemaCollection.Add. This API internally enables DTD processing on the XML reader instance used, and uses UrlResolver for resolving external XML entities. The outcome is information disclosure. Content from file system or network shares for the machine processing the XML can be exposed to attacker. In addition, an attacker can use this as a DoS vector.

File: src/Microsoft.NetCore.Analyzers/UnitTests/Security/DoNotAddSchemaByURLTests.cs [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+using System.Collections.Immutable;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Test.Utilities;
+using Xunit;
+
+namespace Microsoft.NetCore.Analyzers.Security.UnitTests
+{
+    public class DoNotAddSchemaByURLTests : DiagnosticAnalyzerTestBase
```
Comment (dotpaul) [inline src/Microsoft.NetCore.Analyzers/UnitTests/Security/DoNotAddSchemaByURLTests.cs]: A "no diagnostic" unit test that adds to a Dictionary<string, string> would be good.

**Pull request merged**